app/models/nearby_facility.rb

```diff
@@ -7,13 +7,44 @@ class << self
     attr_writer :validate_on_load
 
     def query(params)
-      json_from_file = File.read(Rails.root.join('modules', 'va_facilities', 'nearby.json'))
-      mocked_response = JSON.parse(json_from_file)
-      return mocked_response if params[:street_address] && params[:city] && params[:state] && params[:zip]
+      if params[:street_address] && params[:city] && params[:state] && params[:zip]
+        isochrone_response = request_isochrone(params)
+        return get_facilities_in_isochrone(params, isochrone_response)
+      end
 
       NearbyFacility.none
     end
 
+    def request_isochrone(params)
+      params[:drive_time] = '30' unless params[:drive_time]
+      address = "#{params[:street_address]} #{params[:city]} #{params[:state]} #{params[:zip]}"
+      query = {
+        waypoint: address,
+        maxtime: params[:drive_time],
+        timeUnit: 'minute',
+        dateTime: '07:30:00',
+        optimize: 'timeWithTraffic',
+        key: Settings.bing.key
+      }
+      response = Faraday.get "#{Settings.bing.base_api_url}/Isochrones", query
+      response.body
+    end
+
+    def get_facilities_in_isochrone(params, isochrone_response)
+      isochrone = JSON.parse(isochrone_response)['resourceSets'][0]['resources'][0]['polygons'][0]['coordinates'][0]
```

Comment:
Perhaps throw a comment somewhere with the expected format of the isochrone response?

Reply:
:+1: I do have an example response I can link to.